### DISCUSSION TITLE
Similar crates

### DISCUSSION THREAD

**Discussion by adiba** [rust-lang/crates.io#4331] [Feature Requests]
This is not an easy one:

For the currently viewed crate: Based on the keywords, description and dependencies, one could display a handful of crates ranked by similarity.

Of course you can alter search terms or try through all keywords, but that's tedious.

**Comment by kornelski** [comment]
I've implemented it on crates.rs based on keywords, e.g. https://crates.rs/crates/smallvec shows (in the footer) that it's similar to https://crates.rs/crates/stackvector

The difficulty here is that few crates actually have useful keywords, so the data is very sparse.

**Comment by adiba** [comment]
That's a great start!

Though it doesn't quite work for https://crates.rs/crates/angle and the "angle" keyword.
